### PR TITLE
Deprecate old aliases to generate predicates

### PR DESCRIPTION
**Pull request by dalg24**
This was inconsistent with Boost.Geometry.Index and was only there for "backward compatibility".
I want to have the option reclaim these names, hence the deprecation.

This will be ready when all occurrences are replaced.

### CODE REVIEW

**Comment by aprokop** [comment]
Why not simply remove it? We don't promise backwards compatibility right now, and can simply update DTK.

**Comment by Rombur** [comment]
I agree. It's not like we have users...

**Comment by dalg24** [comment]
Will do.

**Comment by sslattery** [comment]
I agree - just remove

File: src/details/ArborX_Predicates.hpp [hunk 1]
```diff
@@ -66,21 +66,30 @@ struct Intersects
   Geometry _geometry;
 };
 
-using Within = Intersects<Sphere>;
-using Overlap = Intersects<Box>;
+[[deprecated]] using Within = Intersects<Sphere>;
```
Comment (masterleinad) [inline src/details/ArborX_Predicates.hpp]:
```suggestion
using Within [[deprecated]] = Intersects<Sphere>;
```
would work.

**Comment by dalg24** [comment]
This is ready for final review/merge.

**Comment by aprokop** [comment]
@dalg24 Do you want to clean the history?

**Pull request merged**